erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1092,13 +1092,16 @@ def set_billing_hours_and_amount(self):
 					timesheet.billing_amount = ts_doc.total_billable_amount
 
 	def update_timesheet_billing_for_project(self):
-		if not self.timesheets and self.project:
-			self.add_timesheet_data()
-		else:
+		if self.timesheets:
 			self.calculate_billing_amount_for_timesheet()
 
-	@frappe.whitelist()
+	@frappe.whitelist(methods=["PUT"])
 	def add_timesheet_data(self):
+		if not self.timesheets and self.project:
+			self._add_timesheet_data()
+			self.save()
```

Comment:
Why `save`?

Reply:
This is supposed to be called via API. I assume changes would be lost if we don't save?

Reply:
I thought the same function was called fromthe client side